docs/development/typescript.md

```diff
@@ -0,0 +1,64 @@
+# Typescript
+
+**Question**: should we recommend using typescript over javascript?
```

Comment:
My personal feeling is that in general no, but good question for team discussion. I see people split between the "love" or "hate" and I think there are likely contexts where one or the other makes sense. Maybe we can recommend it for some specific contexts?

Reply:
>  I think there are likely contexts where one or the other makes sense. Maybe we can recommend it for some specific contexts?

Exact point. +1

We should avoid love and hate and agree that there are specific contexts for this although they are subjective, e and may end up in long discussions, but is better than nothing.

We can maybe create those contexts/categories/scenarios, for example:

**A** Developer Back End Apps

**B** Performance sensitive projects/libraries

Possible to have that fast project written in TS? an example https://github.com/nodejs/undici

**C** Front End Apps

**D** Tools, CLI tools, libraries, DB drivers, clients (Kafka, Prometheus, etc...)

**E** Team-oriented projects and TS + [clean code](https://github.com/labs42io/clean-code-typescript)

let's face the truth, a lot of JS libraries are written by 1 person only and that is common to avoid SOLID and a lot of other things that are (more) destined for teams.

That may end up with long discussions and some _**"IMO"**_, for example for the item **A** I'm +1 TS